rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -362,24 +371,27 @@ protected void setLRAStatus(int actionStatus) {
     }
 
     boolean isClosed() {
-        return status != null && status.equals(LRAStatus.Closed);
+        assert status != null;
+        return status.equals(LRAStatus.Closed);
```

Comment:
you gave me an idea earlier in the PR :) maybe we can rewrite these comparisons to use `==` as they are enums. We can save some processing.

Reply:
+1

Also `isClosed`, `isCancelled`, `isActive` etc are not required since any caller just needs to call `getStatus()` and then do the test (in fact I think they are unused). Also note that `class Transaction` is internal to the implementation so we have full control over the users of this class.
 
The other methods such as `isFinished` and `isRecovering` are useful since the check multiple states.